common_benchmark_suite/openxla/benchmark/comparative_suite/jax_test_data_definitions.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 The OpenXLA Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import string
+
+from openxla.benchmark.comparative_suite import utils
+from openxla.benchmark import def_types, unique_ids
+
+# Constants and functions help build batch templates.
+BATCH_ID = lambda data_id: string.Template(data_id + "-batch${batch_size}")
+BATCH_NAME = lambda name: string.Template(name + "_BATCH${batch_size}")
+BATCH_TAG = string.Template("batch-${batch_size}")
+BATCH_MODEL_ID = lambda model_id: string.Template(model_id +
+                                                  "-batch${batch_size}")
+BATCH_TENSOR_DIMS = lambda dims: string.Template("${batch_size}x" + dims)
+
+# T5 large inputs.
+T5_LARGE_JAX_SEQLEN512_I32_INPUT_BATCH_TEMPLATE = utils.ModelTestDataTemplate(
+    id=BATCH_ID(unique_ids.INPUT_DATA_T5_LARGE_JAX_SEQLEN512_I32),
+    name=BATCH_NAME("T5_LARGE_JAX_SEQLEN512_I32_INPUT"),
+    tags=["input-data", "seqlen512", BATCH_TAG],
+    source_info=
+    "Original text: 'Studies have been shown that owning a dog is good for you'",
+    artifacts={
+        def_types.ModelTestDataFormat.NUMPY_TENSORS:
+            utils.ModelTestDataArtifactTemplate(
+                data_format=def_types.ModelTestDataFormat.NUMPY_TENSORS,
+                data_parameters={
+                    "tensor_names": ["input_ids", "decoder_input_ids"],
+                    "tensor_dimensions": [
+                        BATCH_TENSOR_DIMS("512xi32"),
+                        BATCH_TENSOR_DIMS("512xi32"),
+                    ]
+                },
+                verify_parameters={},
+                source_url=string.Template(
+                    "https://storage.googleapis.com/iree-model-artifacts/jax/jax_models_0.4.10_1684396752/T5_LARGE/batch_${batch_size}/input_npy.tgz"
+                ))
+    })
+T5_LARGE_JAX_SEQLEN512_I32_INPUT_BATCHES = utils.build_batch_model_test_data(
+    template=T5_LARGE_JAX_SEQLEN512_I32_INPUT_BATCH_TEMPLATE,
+    batch_sizes=[1, 16, 24, 32, 48, 64, 512])
+
+# T5 large outputs.
+T5_LARGE_FP32_JAX_512X1024XF32_OUTPUT_BATCH_TEMPLATE = utils.ModelTestDataTemplate(
+    id=BATCH_ID(unique_ids.OUTPUT_DATA_T5_LARGE_FP32_JAX_512X1024XF32),
+    name=BATCH_NAME("T5_LARGE_FP32_JAX_512X1024XF32_OUTPUT"),
+    tags=["output-data", BATCH_TAG],
+    source_info="",
+    artifacts={
+        def_types.ModelTestDataFormat.NUMPY_TENSORS:
+            utils.ModelTestDataArtifactTemplate(
+                data_format=def_types.ModelTestDataFormat.NUMPY_TENSORS,
+                data_parameters={
+                    "tensor_names": ["output_0"],
+                    "tensor_dimensions": [BATCH_TENSOR_DIMS("512x1024xi32"),]
+                },
+                # TODO(#11): Add verification tolerance.
+                verify_parameters={},
```

Comment:
I've used 0.5 so far. It's quite a high tolerance since we're generating output data on CPU and testing against outputs generated on GPU. On GPU, we use TF32 data type which won't be bit-exact.

Reply:
Doc in the TODO comment. We will implement this later.